test/integration/commands/bootstrap.ts

```diff
@@ -69,21 +61,31 @@ const teardown = async ({ apiKey, environments }: { apiKey?: ApiKey; environment
   await Promise.allSettled([apiKey && apiKey.delete(), ...environments.map((env) => env.delete())])
 }
 
-export const createEnvironments = async (testSpace: Space): Promise<TestContext> => {
-  const sourceEnvironment = await testUtils.createTestEnvironment(testSpace, randomId + 'source_for_create')
-  const targetEnvironment = await testUtils.createTestEnvironment(testSpace, randomId + 'target_for_create')
-  const apiKey = await createCdaToken(testSpace, [targetEnvironment.sys.id, sourceEnvironment.sys.id])
-
-  return {
-    sourceEnvironment,
-    targetEnvironment,
-    cdaToken: apiKey.accessToken,
-    spaceId: testSpace.sys.id,
-    teardown: () => teardown({ apiKey, environments: [sourceEnvironment, targetEnvironment] }),
+export const createEnvironments = async (testSpace: Space): Promise<TestContext | undefined> => {
+  try {
+    const randomId = Math.floor(Math.random() * 10000).toString() // used to prevent concurrency issues
+    console.log('creating source environment...')
+    const sourceEnvironment = await testUtils.createTestEnvironment(testSpace, randomId + '_source_environment')
+    console.log('creating target environment...')
+    const targetEnvironment = await testUtils.createTestEnvironment(testSpace, randomId + '_target_environment')
+    console.log('creating API keys...')
+    const apiKey = await createCdaToken(testSpace, [targetEnvironment.sys.id, sourceEnvironment.sys.id])
+    console.log('all setup')
+    return {
+      sourceEnvironment,
+      targetEnvironment,
+      cdaToken: apiKey.accessToken,
+      spaceId: testSpace.sys.id,
+      teardown: () => teardown({ apiKey, environments: [sourceEnvironment, targetEnvironment] }),
+    }
+  } catch (error) {
+    console.error(error)
```

Comment:
why catch it here and then rethrow in the other file (see the other comment)?

Reply:
Good point, I think it's like this for the debugging, I'll adjust it 👍